MaterialDesignThemes.Wpf/Behaviors/PasswordBoxRevealTextBoxBehavior.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System.Reflection;
+using System.Windows.Documents;
+using Microsoft.Xaml.Behaviors;
+
+namespace MaterialDesignThemes.Wpf.Behaviors;
+
+internal class PasswordBoxRevealTextBoxBehavior : Behavior<TextBox>
+{
+    private static readonly DependencyProperty SelectionProperty = DependencyProperty.RegisterAttached(
+        "Selection", typeof(TextSelection), typeof(PasswordBoxRevealTextBoxBehavior), new UIPropertyMetadata(default(TextSelection)));
+    private static void SetSelection(DependencyObject obj, TextSelection? value) => obj.SetValue(SelectionProperty, value);
+    private static TextSelection? GetSelection(DependencyObject obj) => (TextSelection?)obj.GetValue(SelectionProperty);
+
+    internal static readonly DependencyProperty PasswordBoxProperty = DependencyProperty.Register(
+        nameof(PasswordBox), typeof(PasswordBox), typeof(PasswordBoxRevealTextBoxBehavior), new PropertyMetadata(default(PasswordBox)));
+
+    internal PasswordBox? PasswordBox
+    {
+        get => (PasswordBox) GetValue(PasswordBoxProperty);
+        set => SetValue(PasswordBoxProperty, value);
+    }
+
+    protected override void OnAttached()
+    {
+        base.OnAttached();
+        AssociatedObject.IsVisibleChanged += AssociatedObjectOnIsVisibleChanged;
+        if (PasswordBox != null)
+        {
+            var info = typeof(PasswordBox).GetProperty("Selection", BindingFlags.NonPublic | BindingFlags.Instance);
+            var selection = info?.GetValue(PasswordBox, null) as TextSelection;
+            SetSelection(AssociatedObject, selection);
+        }
+    }
+
+    protected override void OnDetaching()
+    {
+        base.OnDetaching();
+        if (AssociatedObject != null)
+        {
+            AssociatedObject.ClearValue(SelectionProperty);
+            AssociatedObject.IsVisibleChanged -= AssociatedObjectOnIsVisibleChanged;
+        }
+    }
+
+    private void AssociatedObjectOnIsVisibleChanged(object sender, DependencyPropertyChangedEventArgs e)
+    {
+        if (AssociatedObject.IsVisible)
+        {
+            AssociatedObject.SelectionLength = 0;
+            var selection = GetPasswordBoxSelection();
+            AssociatedObject.SelectionStart = selection.SelectionStart;
+            AssociatedObject.SelectionLength = selection.SelectionEnd;
+            Keyboard.Focus(AssociatedObject);
+        }
+        else if (PasswordBox != null)
+        {
+            SetPasswordBoxSelection(AssociatedObject.SelectionStart, AssociatedObject.SelectionLength);
+            Keyboard.Focus(PasswordBox);
+        }
+    }
+
+    private PasswordBoxSelection GetPasswordBoxSelection()
+    {
+        var selection = GetSelection(AssociatedObject);
+        var typeTextRange = selection?.GetType().GetInterfaces().FirstOrDefault(i => i.Name == "ITextRange");
```

Comment:
I would also pull the reflection look-ups here out into statics to avoid duplicate reflection calls across multiple instances (may be helpful to add a static constructor to allow for some more logic). 

For example, because this one is targeting a specific interface I would extract the interface by name rather than by going off of the `selection.GetType()`.
Something like this:
```C#
static PasswordBoxRevealTextBoxBehavior()
{
    Type iTextRange = typeof(PasswordBox).Assembly.GetType("System.Windows.Documents.ITextRange")
        ?? throw new InvalidOperationException("Failed to find ITextRange");
    MethodInfo getStart = iTextRange.GetProperty("Start")?.GetGetMethod()
        ?? throw new InvalidOperationException($"Failed to find Start property on {iTextRange.FullName}");
        ...
}
```

Reply:
Sounds reasonable. Will change.